common/src/main/java/com/wynntils/models/players/GuildModel.java

```diff
@@ -87,12 +112,25 @@ public class GuildModel extends Model {
     // Test in GuildModel_OBJECTIVE_STREAK_PATTERN
     private static final Pattern OBJECTIVE_STREAK_PATTERN = Pattern.compile("^§a- §7Streak: §f(?<streak>\\d+)$");
 
+    // Test in GuildModel_TRIBUTE_PATTERN
+    private static final Pattern TRIBUTE_PATTERN = Pattern.compile(
+            "^§[abef6](?<symbol>[ⒷⒸⓀⒿ]?) ?(?<amount>[+-]\\d+) (Ore|Wood|Fish|Crops?|Emeralds?) per Hour$");
+
+    // Test in GuildModel_ALLIED_GUILD_PATTERN
+    private static final Pattern ALLIED_GUILD_PATTERN =
+            Pattern.compile("§a- §7(?<name>[a-zA-Z\\s]+) \\[[a-zA-Z]{3,4}\\]");
+
     private static final int MEMBERS_SLOT = 0;
     private static final int OBJECTIVES_SLOT = 13;
+    private static final int DIPLOMACY_MENU_SLOT = 26;
+    private static final List<Integer> DIPLOMACY_SLOTS = List.of(2, 3, 4, 5, 6, 7, 8);
 
     private static final List<Integer> OBJECTIVE_GOALS = List.of(5, 15, 30);
 
     private Map<String, GuildProfile> guildProfileMap = new HashMap<>();
+    // sent, received
+    private final Map<String, Pair<Map<GuildResource, Integer>, Map<GuildResource, Integer>>> guildDiplomacyMap =
```

Comment:
It may be better to create a holder class/record for this, and "name" the objects in the pairs. That way you can do `guildDiplomacyMap.getGuild("ANO").getSent()`, which is much more verbose/readable than it is currently.